fifo-animal-shelter/fifo-animal-shelter.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+module.exports = class AnimalShelter {
+  constructor() {
+    this.animals = [];
+  }
+
+  enqueue(animal) {
+    this.animals.push(animal);
+  }
+
+  dequeue(pref) {
+    let returned = null;
+    if (pref) {
+      for (let x = 0; x < this.animals.length; x++) {
```

Comment:
No points knocked off for this, but remember what a `queue` data structure allows. It's FIFO and only has enqueue and dequeue methods attached to it. You could technically traverse it with a for loop like you're doing, BUT if you're needing to traverse, the data structure shouldn't really be considered a queue anymore.

In other words, if we know a queue is `FIFO`, and we only ever need to worry about the data that's first in the queue, is there another way to go through this queue rather than a for loop?